src/libs/globbing/globbing.c

```diff
@@ -0,0 +1,212 @@
+/**
+ * @file
+ *
+ * @brief Library for performing globbing on keynames.
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include <kdb.h>
+#include <kdbglobbing.h>
+#include <kdbhelper.h>
+
+#include <ctype.h>
+#include <fnmatch.h>
+#include <stdlib.h>
+#include <string.h>
+
+static size_t strcnt (const char * s, char c)
+{
+	size_t count = 0;
+	while ((s = strchr (s + 1, c)) != NULL)
+	{
+		count++;
+	}
+	return count;
+}
+
+static char * elektraToFnmatchGlob (char * pattern)
+{
+	char * ptr = pattern;
+	while ((ptr = strchr (ptr, '/')) != NULL)
+	{
+		++ptr;
+		if ((*(ptr + 1) == '/' || *(ptr + 1) == '\0') && (*ptr == '#' || *ptr == '_'))
+		{
+			*ptr = '*'; // replace /#/ and /_/ with /*/
+		}
+	}
+	return pattern;
+}
+
+/**
+ * @brief checks whether the given string is a valid elektra array item name
+ *
+ * Valid array items consist of a '#' followed by <code>n</code> underscores ('_'),
+ * followed by <code>n+1</code> digits ('0'-'9'). Additionally the digits describe a
+ * valid 32-bit Integer (i.e. <code>0 <= x < 2^32</code>).
+ *
+ * @param name the string to check
+ * @retval true if @p name is a valid array item
+ * @retval false otherwise
+ */
+bool isArrayName (const char * name) // TODO: move? definitely useful elsewhere, e.g. reference plugin, libease, etc.
```

Comment:
Isn't this the same as elektraArrayValidateName from libease?

If not, putting this to libease makes more sense.

Reply:
Yes, this is the basically the same as in libease, but I wasn't sure whether a dependency on libease just for this functionality was a good idea. If the libglobbing depends on libease, then I don't see the advantage in having a separate libglobbing at all.

Personally I think that this function should be part of the core, because elektraWriteArrayNumber` is also declared in `kdbhelper.h` and implemented in `keyset.c`

Reply:
> If the libglobbing depends on libease, then I don't see the advantage in having a separate libglobbing at all.

libglobbing uses POSIX, libease not.

> Personally I think that this function should be part of the core

It quite clearly should not be. The core is only for functionality that directly depends on internal structures. It should be as minimal as possible.

> elektraWriteArrayNumber

You are right, elektraWriteArrayNumber should not be in the core. I hope d95edabb20264aef6db35085ad329d43b70e9735 makes it more clear.